net/src/main/java/io/bitcoinsv/jcl/net/protocol/handlers/discovery/DiscoveryHandlerConfig.java

```diff
@@ -223,6 +243,16 @@ public DiscoveryHandlerConfig.DiscoveryHandlerConfigBuilder addInitialConnection
             return this;
         }
 
+        public DiscoveryHandlerConfig.DiscoveryHandlerConfigBuilder addrBroadcastFrequency(Optional<Duration> addrBroadcastFrequency) {
+            this.addrBroadcastFrequency = addrBroadcastFrequency;
+            return this;
+        }
+
+        public DiscoveryHandlerConfig.DiscoveryHandlerConfigBuilder advertisedIp(Optional<String> advertisedIp){
+            this.advertisedIp = advertisedIp;
```

Comment:
Add back the check for empty string, otherwise if the variable is unset in .env file, it will be set to empty string here.

Reply:
I suggest we keep this definition so it is consistent with other methods defined in this class. Besides, empty string will cause an exception in `DiscoveryHandlerImpl`'s `jobBroadcastOwnAddrMsg` where the `PeerAddress.fromIp(advertisedIp.get())` is called and the exception will be properly handled inside the try-catch block.

Reply:
Resolving conversation: As discussed, the empty string issue from .env file is not part of JCL and is an issue of the application/project that is configuring the JCL P2P component.